pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -313,6 +313,8 @@ def _prepare_dataloader(
             sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
             dataloader = _update_dataloader(dataloader, sampler, mode=mode)
 
+        dataloader = self.trainer.strategy.process_dataloader(dataloader)
```

Comment:
(I might be wrong here) - Do you think it's odd to call a public method from a assumably private method? I know private methods in Python are just those which start with `_` but still, asking out of curiosity.

Reply:
Good callout~ Maybe code smell? But I think `DataConnector` has `trainer` property, which means we can get access to any public functions through `trainer`.  I think it's fine to call the public function from `Strategy` class, as long as it's not public function from `DataConnector`.  We need revisit this `_prepare_dataloader`, as I'm not sure why it's private func. It has been used for reset train/val dataloader frequently.

Reply:
Thanks, @ninginthecloud for helping with the explanation behind this. Agreed that we can revisit the `_prepare_dataloader`, but let's not block this PR and get it landed if it looks good for everyone else as well. 🚀